app/services/claims/claim_window/build.rb

```diff
@@ -0,0 +1,30 @@
+class Claims::ClaimWindow::Build
+  include ServicePattern
+
+  def initialize(claim_window_params:, claim_window: nil)
+    @claim_window = claim_window
+    @claim_window_params = claim_window_params
+  end
+
+  def call
+    self.claim_window ||= Claims::ClaimWindow.new
+    self.claim_window.assign_attributes(claim_window_params)
+    self.claim_window.academic_year = find_academic_year
+    self.claim_window
+  end
+
+  private
+
+  attr_accessor :claim_window, :claim_window_params
+
+  def find_academic_year
+    return unless claim_window.starts_on
+
+    year = claim_window.starts_on.month > 1 ? claim_window.starts_on.year - 1 : claim_window.starts_on.year - 2
+
+    AcademicYear.create_with(
+      starts_on: Date.new(year, 8, 1),
+      ends_on: Date.new(year + 1, 7, 31),
```

Comment:
I'm uncomfortable with the these [magic numbers](https://en.wikipedia.org/wiki/Magic_number_(programming)) here, which are also hardcoded into the data migration.